opentelemetry-sdk/src/metrics/meter.rs

```diff
@@ -75,14 +76,20 @@ impl SdkMeter {
     {
         let validation_result = validate_instrument_config(builder.name.as_ref(), &builder.unit);
         if let Err(err) = validation_result {
-            global::handle_error(err);
+            otel_error!(
+                name: "InstrumentCreationFailed",
+                meter_name = self.scope.name.as_ref(),
+                instrument_name = builder.name.as_ref(),
+                message = "Measurements from this counter will be ignored.",
+                reason = format!("{}", err)
+            );
```

Comment:
2024-10-24T05:32:23.524520Z ERROR main opentelemetry_sdk meter_name="mylibraryname" instrument_name="-my_counter" Measurements from this counter will be ignored. reason="Invalid instrument configuration: instrument name must start with an alphabetic character"

^ This is what we get when I used fmt module. The reason is a slightly extra verbose:
Current: _Invalid instrument configuration: instrument name must start with an alphabetic character_
Proposal: _Instrument name must start with an alphabetic character_

This is a minor thing we can change in the future.